src/core/CCParse.ml

```diff
@@ -87,7 +87,7 @@ let next st ~ok ~err =
   else (
     let c = st.str.[st.i] in
     st.i <- st.i + 1;
-    if c='\n'
+    if CCChar.equal c '\n'
```

Comment:
in this module, again, too many cross dependencies. What's wrong with `(=)` on strings and chars?